qiskit/chemistry/qubit_transformations/fermionic_transformation.py

```diff
@@ -0,0 +1,571 @@
+                operator symmetry. If it does not then the operator will be discarded since no
+                meaningful measurement can take place.
+        Raises:
+            QiskitChemistryError: Invalid symmetry reduction
+        """
+        transformation = transformation.value
+        orbital_reduction = orbital_reduction if orbital_reduction is not None else []
+        super().__init__()
+        self._transformation = transformation
+        self._qubit_mapping = qubit_mapping.value
+        self._two_qubit_reduction = two_qubit_reduction
+        self._freeze_core = freeze_core
+        self._orbital_reduction = orbital_reduction
+        if z2symmetry_reduction is not None:
+            if isinstance(z2symmetry_reduction, str):
+                if z2symmetry_reduction != 'auto':
+                    raise QiskitChemistryError('Invalid z2symmetry_reduction value')
+        self._z2symmetry_reduction = z2symmetry_reduction
+        self._has_dipole_moments = False
+
+        # Store values that are computed by the classical logic in order
+        # that later they may be combined with the quantum result
+        self._hf_energy = None
+        self._nuclear_repulsion_energy = None
+        self._nuclear_dipole_moment = None
+        self._reverse_dipole_sign = None
+        # The following shifts are from freezing orbitals under orbital reduction
+        self._energy_shift = 0.0
+        self._x_dipole_shift = 0.0
+        self._y_dipole_shift = 0.0
+        self._z_dipole_shift = 0.0
+        # The following shifts are from particle_hole transformation
+        self._ph_energy_shift = 0.0
+        self._ph_x_dipole_shift = 0.0
+        self._ph_y_dipole_shift = 0.0
+        self._ph_z_dipole_shift = 0.0
+
+        self._molecule_info: Dict[str, Any] = {}
+
+    @property
+    def molecule_info(self) -> Dict[str, Any]:
+        """Getter of the molecule information."""
+        return self._molecule_info
+
+    @property
+    def qubit_mapping(self) -> str:
+        """Getter of the qubit mapping."""
+        return self._qubit_mapping
+
+    def transform(self, driver: BaseDriver,
+                  aux_operators: Optional[List[FermionicOperator]] = None
+                  ) -> Tuple[WeightedPauliOperator, List[WeightedPauliOperator]]:
+        """Transformation from the ``driver`` to a qubit operator.
+
+        Args:
+            driver: A driver encoding the molecule information.
+            aux_operators: Additional auxiliary ``FermionicOperator``s to evaluate.
+
+        Returns:
+            A qubit operator and a dictionary of auxiliary operators.
+        """
+        q_molecule = driver.run()
+        ops, aux_ops = self._do_transform(q_molecule, aux_operators)
+
+        return ops, aux_ops
+
+    def _do_transform(self, qmolecule: QMolecule,
+                      aux_operators: Optional[List[Union[FermionicOperator,
+                                                         WeightedPauliOperator]]] = None
```

Comment:
Why does this list include WPO - its an internal private method and so far the caller, just above can only be FermionicOperators, which it directly passes on.

Reply:
@Cryoris please correct me if I am wrong. But I believe we did this in a way that one can pass directly a list of aux_operators of WPO type to make it more accessible and easier to use.

Reply:
Support removed in 850c79975, as no longer necessary due to `evaluate_operators`